### PR TITLE
Minor Improvements

### PR DESCRIPTION
**Pull request by arturobernalg**
* Add final
* Remove redundant initializer
* Replace String in loop by StringBuilder
* Use Empty  array

### CODE REVIEW

File: src/main/java/org/apache/commons/daemon/support/DaemonWrapper.java [hunk 1]
```diff
@@ -40,12 +40,17 @@
...
     private final static String STOP_ARGS       = "stop.args";
-    private String              configFileName  = null;
+    private String              configFileName;
     private final DaemonConfiguration config;
 
     private final Invoker             startup;
     private final Invoker             shutdown;
 
+    /**
+     * An empty immutable {@code String} array.
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/daemon/support/DaemonWrapper.java:50]: This does not need to be public does it?

File: src/main/java/org/apache/commons/daemon/support/DaemonConfiguration.java [hunk 2]
```diff
@@ -62,6 +62,11 @@
     private final Properties configurationProperties;
     private final Properties systemProperties;
 
+    /**
+     * An empty immutable {@code String} array.
+     */
```
Comment (garydgregory) [inline src/main/java/org/apache/commons/daemon/support/DaemonConfiguration.java:67]: This does not need to be public. You could define ONE constant to be package private and reuse it.

**Comment by garydgregory** [comment]
Also please rebase on master to pick up the GitHub build addition.

File: src/main/java/org/apache/commons/daemon/support/DaemonWrapper.java (hunk 1 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/daemon/support/DaemonWrapper.java]: Changed

File: src/main/java/org/apache/commons/daemon/support/DaemonConfiguration.java (hunk 2 above)
Comment (arturobernalg) [reply src/main/java/org/apache/commons/daemon/support/DaemonConfiguration.java:67]: Changed

File: src/test/java/org/apache/commons/daemon/SimpleDaemon.java [hunk 3]
```diff
@@ -28,18 +28,18 @@
...
+    private volatile boolean stopping;
+    private String directory;
+    private final Vector<Handler> HANDLERS;
     private boolean softReloadSignalled;
 
     public SimpleDaemon() {
         System.err.println("SimpleDaemon: instance "+this.hashCode()+
                            " created");
-        this.handlers=new Vector<Handler>();
+        this.HANDLERS =new Vector<Handler>();
```
Comment (garydgregory) [inline src/test/java/org/apache/commons/daemon/SimpleDaemon.java:42]: Why is this upper case?

Comment (arturobernalg) [reply src/test/java/org/apache/commons/daemon/SimpleDaemon.java]: changed

Comment (garydgregory) [reply src/test/java/org/apache/commons/daemon/SimpleDaemon.java]: It is still upper case.

**Review by garydgregory — APPROVED** [review]

**Pull request merged**